zookeeper-server/src/test/java/org/apache/zookeeper/test/BackupManagerTest.java

```diff
@@ -0,0 +1,601 @@
+    Assert.assertEquals("backupStatus should be set to min for log",
+        bp.getLogZxid(), BackupUtil.INVALID_LOG_ZXID);
+    Assert.assertEquals("backupStatus should be set to min for snap",
+        bp.getSnapZxid(), BackupUtil.INVALID_SNAP_ZXID);
+  }
+
+  @Test
+  public void testSwitchToNewBackupNode() throws Exception {
+    createNodes(connection, 1000);
+    backupManager.getLogBackup().run(1);
+    backupManager.getSnapBackup().run(1);
+
+    BackupPoint bp = backupStatus.read();
+    String[] filesOrig = getBackupFiles(backupDir);
+    Assert.assertEquals("backup status and backupmanager must match log",
+        bp.getLogZxid(), backupManager.getBackedupLogZxid());
+    Assert.assertEquals("backup status and backupmanager must match snap",
+        bp.getSnapZxid(), backupManager.getBackedupSnapZxid());
+
+    File backupStatusFile = new File(backupStatusDir, BackupStatus.STATUS_FILENAME);
+    backupStatusFile.delete();
+
+    backupManager.initialize();
+    BackupPoint bp2 = backupStatus.read();
+    Assert.assertEquals(
+        "backup status and backupmanager must match log after switch to new backup node",
+        bp2.getLogZxid(),
+        backupManager.getBackedupLogZxid());
+    Assert.assertEquals(
+        "backup status and backupmanager must match snap after switch to new backup node",
+        bp2.getSnapZxid(),
+        backupManager.getBackedupSnapZxid());
+    Assert.assertEquals("backup point must match prior backup point -- snap",
+        bp.getLogZxid(), bp2.getLogZxid());
+    Assert.assertEquals("backup point must match prior backup point -- snap",
+        bp.getSnapZxid(), bp2.getSnapZxid());
+
+    backupManager.getLogBackup().run(1);
+    backupManager.getSnapBackup().run(1);
+
+    String[] filesPost = getBackupFiles(backupDir);
+    Assert.assertArrayEquals("files before and after switch to new backup node",
+        filesOrig, filesPost);
+
+    createNodes(connection, 1000);
+    backupManager.getLogBackup().run(1);
+    backupManager.getSnapBackup().run(1);
+
+    String[] filesPostCreates = getBackupFiles(backupDir);
+    List<Range<Long>> zxids = BackupUtil.getZxids(backupStorage, BackupFileType.TXNLOG);
+
+    Assert.assertTrue("must have more backups", filesPostCreates.length > filesPost.length);
+
+    for (int i = 0; i < zxids.size(); i++) {
+      LOG.info("zxids[{}]: {}", i, zxids.get(i));
+    }
+
+    for (int i = 0; i < zxids.size() - 1; i++) {
+      Assert.assertEquals("max of previous must be one less than next min",
+          (zxids.get(i).upperEndpoint() + 1), zxids.get(i + 1).lowerEndpoint().longValue());
+    }
+  }
+
+  @Test
+  public void testBackupManager() throws Exception {
+    backupManager.initialize();
+
+    backupManager.getSnapBackup().run(1);
+
+    String[] files = getBackupFiles(backupDir);
+    Assert.assertTrue("Must have some snapshots", files != null);
+
+    Assert.assertEquals("Must have just a single snapshot", 1, files.length);
```

Comment:
Do we need to reset the directories before this test because they may already have backup files from the other tests, so this assertion won't fail? Same applies to the other tests.

Reply:
I think `backupManager.initialize();` should do what you described.